Automattic/simplenote-ios
# Siri Shortcuts

### Details:
We're adding support for Siri Shortcuts for three actions (initially!):

1. Launch Simplenote
2. Add a new Note
3. Open a specific Note


@bummytime ready for a look!! thanks in advance!!
Closes #236 

### NOTE:

Exposing UserActivities by means of **becomeCurrent** fails miserably, when exposing multiple activities at once (as per... tests with the actual device). iOS either picks up the last registered activity, or none of them!


---

### Testing: Spotlight Search Sanity
1. Log into your account
2. Send Simplenote to Background
3. Search in Spotlight the contents of **any** of your notes, and tap on any of them
4. Verify Simplenote is brought to FG, and the note is onscreen

### Testing: Opening Simplenote
1. Open **Settings.app** > **Deverloper** and enable `Display recent Shortcuts`
2. Log into your account
3. Send Simplenote to BG
4. Open Spotlight
5. Verify `Open Simplenote` shows up. Tap on it, and verify Simplenote is brought to FG


### Testing: Creating a New Note
1. Open **Settings.app** > **Deverloper** and enable `Display recent Shortcuts`
2. Log into your account
3. Create a new Note, so that the new shortcut is registered
4. Open Spotlight
5. Verify `Create a new note ` shows up. Tap on it, and verify Simplenote is brought up onscreen with a blank note

### Testing: Opening a Specific Note
1. Open **Settings.app** > **Deverloper** and enable `Display recent Shortcuts`
2. Log into your account
3. Open any random note
4. Open Spotlight
5. Verify an item named `Open [Title of your note]` shows up. Tap on it, and verify the note you've opened in (3) is onscreen
